### PR TITLE
CF:Identifiedparticles

### PR DESCRIPTION
**Pull request by sweta29singh**
Mean pT fluctuation in identified particle on run 3 data

### CODE REVIEW

File: PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt [hunk 1]
```diff
@@ -19,3 +19,9 @@ o2physics_add_dpl_workflow(mean-pt-fluc-id
                     PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore O2Physics::PWGCFCore
                     COMPONENT_NAME Analysis)
 
+
+o2physics_add_dpl_workflow(mean-pt-pbpb-fluc
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt:23]: Please, don't use the system (`pbpb`) in the name
At some point the system on which the analysis might run will be a configurable, i.e. the analysis will be able to run on `pp` or `PbPb` just changing a configurable parameter

As this is an identified particle analysis please make that explicit in the executable name and also in the name of the source file

File: PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \author Sweta Singh (sweta.singh@cern.ch)
+
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx:13]: Please consider my above comment regarding the name of the source file

File: PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+#include "Framework/HistogramRegistry.h"     //
+#include "Framework/RunningWorkflowInfo.h"   //
+#include "CommonConstants/MathConstants.h"   //
+#include "PWGCF/Core/CorrelationContainer.h" //
+#include "PWGCF/Core/PairCuts.h"             //
+
+#include "Common/DataModel/FT0Corrected.h" //
+// #include "Framework/ASoA.h"
+// #include "TRandom.h"//
+// #include "/home/sweta/alice/sw/ubuntu2004_x86-64/O2Physics/master-local5/include/TrackSelectionTables.h"
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx:38]: Please, don't leave code commented
Please, consider all appearances

File: PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+                           aod::pidTPCFullKa, aod::pidTPCFullPr, aod::pidTPCFullDe,
+                           aod::pidTOFFullEl, aod::pidTOFFullMu, aod::pidTOFFullPi, aod::StoredTracks,
+                           aod::pidTOFFullKa, aod::pidTOFFullPr, aod::pidTOFFullDe, aod::pidTOFbeta, aod::TOFSignal,
+                           aod::TracksExtra, aod::TracksIU, aod::TracksDCA, aod::TrackSelection>;
+
+using MyCollision = MyCollisions::iterator;
+using MyTrack = MyTracks::iterator;
+} // namespace o2::aod
+
+struct MeanpTFlucPbPbIdentified {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx:70]: As above mentioned, please, remove `PbPb` from the name

File: PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+
+    int sample = histos.get<TH1>(HIST("hZvtx_before_sel"))->GetEntries();
+    sample = sample % 30;
+
+    // Perfroming the track selection==============================================================================================================
+    for (auto track : inputTracks) { // Loop over tracks
+
+      //  if (!track.isGlobalTrack()) return;
+
+      if (!((fabs(track.eta()) < 0.8) && (fabs(track.dcaXY()) < 0.12) && (fabs(track.dcaZ()) < 1.) && (track.pt() > 0.15 && track.pt() < 2.))) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx:241]: Consider making benefit of the filtering

File: PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+          histos.fill(HIST("hPtPion"), track.pt());
+          histos.fill(HIST("hEtaPion"), track.eta());
+
+          VMeanPtPion.push_back(track.pt());
+
+          for (ULong64_t jPi = 0; jPi < VMeanPtPion.size(); jPi++) {
+            histos.fill(HIST("hPtChPion"), nCh, VMeanPtPion[jPi]);
+          }
+
+          VMeanPtPion.clear();
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx:306]: Consider you are doing this for every pion and you clear the vector after doing it
Is that what you wanted?

The same comment apply for the case of kaons and protons, I guess

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please, consider my renaming comments and also removing the commented code

File: PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt (hunk 1 above)
Comment (victor-gonzalez) [reply PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt:23]: If you push your new commits to your gitHub repo the PR will get automatically updated

Comment (sweta29singh) [reply PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt:23]: Thank you Victor for the comments. I will modify the task according to your suggestions. 
After modifying the comments do i need to create a new pull request again with the modify code and CMakeList.txt?

> If you push your new commits to your gitHub repo the PR will get automatically updated

Thanks Victor. I will push new commits by today.

Comment (sweta29singh) [reply PWGCF/EbyEFluctuations/Tasks/CMakeLists.txt]: > If you push your new commits to your gitHub repo the PR will get automatically updated 

Hii Victor,
I have push my new commit in the branch MeanpTPID. I am getting some error in MegaLinter. Could you please guide me what could be the issue?
Thanks in Advance!
Regards
Sweta

**Review by victor-gonzalez — DISMISSED** [review]

File: PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+        if ((track.hasTPC() && (track.pt() >= 0.5 && track.pt() < 2.0)) && (track.hasTOF() && abs(sqrt(track.tpcNSigmaPi()) * (track.tpcNSigmaPi()) + (track.tofNSigmaPi()) * (track.tofNSigmaPi())) < 2.)) {
+
+          histos.fill(HIST("hPtPion"), track.pt());
+          histos.fill(HIST("hEtaPion"), track.eta());
+
+          VMeanPtPion.push_back(track.pt());
+
+          for (ULong64_t jPi = 0; jPi < VMeanPtPion.size(); jPi++) {
+            histos.fill(HIST("hPtChPion"), nCh, VMeanPtPion[jPi]);
+          }
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx:280]: Are you sure you want to do this on a per pion track basis?
Same comment apply below for kaons and  protons

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please check my comment about pions, kaons and protons

File: PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+
+using MyCollision = MyCollisions::iterator;
+using MyTrack = MyTracks::iterator;
+} // namespace o2::aod
+
+struct IdentifiedMeanPtFluctuations {
+  HistogramRegistry histos{"Histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  // Equivalent of the AliRoot task UserCreateOutputObjects
+  void init(o2::framework::InitContext&)
```
[the remark this answers is not in the document]
Comment (sweta29singh) [reply PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx]: > Please, get rid of this comment AliRoot does not exist in the O2 context

Dear Victor,

I remove most of the comment(all related to AliROOT kind of thing) from my code, it only contains some comment line for the sake of my understanding. I tried to add, commit and push many times but everytime I am getting error in Megalinter. When I check the error message in MegaLinter it is showing here below. Please suggest something regarding this issue. 

Thanks in Advance!
With Regards
Sweta

  **--Error detail:**
  PWGLF/TableProducer/lfTPCPID.cxx:68:  For a static/global string constant, use a C style string instead: "const char defaultOptions[]".  [runtime/string] [4]
  PWGLF/TableProducer/lfTPCPID.cxx:81:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
  PWGLF/Tasks/lambdakzeroanalysis.cxx:42:  Found C system header after other header. Should be: lambdakzeroanalysis.h, c system, c++ system, other.  [build/include_order] [4]
  PWGLF/Tasks/lambdakzeroanalysis.cxx:45:  Found C++ system header after other header. Should be: lambdakzeroanalysis.h, c system, c++ system, other.  [build/include_order] [4]
  PWGLF/Tasks/lambdakzeroanalysis.cxx:46:  Found C++ system header after other header. Should be: lambdakzeroanalysis.h, c system, c++ system, other.  [build/include_order] [4]
  PWGLF/Tasks/lambdakzeroanalysis.cxx:47:  Found C++ system header after other header. Should be: lambdakzeroanalysis.h, c system, c++ system, other.  [build/include_order] [4]
  PWGLF/Tasks/lambdakzeroanalysis.cxx:48:  "Framework/ASoAHelpers.h" already included at PWGLF/Tasks/lambdakzeroanalysis.cxx:26  [build/include] [4]

[the remark this answers is not in the document]
Comment (victor-gonzalez) [reply PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx]: You can comment whatever you consider needs to be commented but don't refer to something that does not exists in this context

Regarding the MegaLinter error, please rebase your changes to the latest version of O2Physics

[the remark this answers is not in the document]
Comment (sweta29singh) [reply PWGCF/EbyEFluctuations/Tasks/IdentifiedMeanPtFluctuations.cxx]: > You can comment whatever you consider needs to be commented but don't refer to something that does not exists in this context
> 
> Regarding the MegaLinter error, please rebase your changes to the latest version of O2Physics

Thanks for your quick reply. I will fix this thing by rebase changes.

**Comment by victor-gonzalez** [comment]
You still have conflicts which end with the message you got and the `MegaLinter` errors
You have to `rebase` your changes to the latest version of O2Physics

**Comment by sweta29singh** [comment]
> You still have conflicts which end with the message you got and the `MegaLinter` errors You have to `rebase` your changes to the latest version of O2Physics

Dear Victor
Thanks for your comment and suggestions. it keeps me motivated to the right track. Now All checks are done and MegaLinetr errors issue is also done. Please check this code so that I can do this analysis on Hyperloop.
Thanks and Regards
Sweta

**Comment by victor-gonzalez** [comment]
You still are getting the message `This branch has conflicts that must be resolved` which means you are still not using the latest version of `O2Physics`
You have to rebase your changes to the latest version of `O2Physics`

**Comment by sweta29singh** [comment]
> You still are getting the message `This branch has conflicts that must be resolved` which means you are still not using the latest version of `O2Physics` You have to rebase your changes to the latest version of `O2Physics`

Dear Victor,
Now All conflicts has been resolved and all checks are Done. Thanks for your guidance.
Please have a look on it once again. 
Kindly merged this code so that I can use it for analysis.
With Regards
Sweta

**Comment by sweta29singh** [comment]
> I approve it but you have not considered my comment on lines 266-268 which also applies to the other species. I would encourage you to check that what it is in the code is what you really want



> I approve it but you have not considered my comment on lines 266-268 which also applies to the other species. I would encourage you to check that what it is in the code is what you really want

I will look into the code again for pion, kaon and proton species. And will modify this in the code.

**Pull request merged**